src/include/sof/audio/audio_stream.h

```diff
@@ -491,9 +491,8 @@ static inline void audio_stream_produce(struct audio_stream __sparse_cache *buff
 	buffer->w_ptr = audio_stream_wrap(buffer,
 					  (char *)buffer->w_ptr + bytes);
```

Comment:
@singalsu can you add more details about the which case which triggered this patch?

Reply:
I was reading code when debugging other issue. I don't understand buffer produce needs to touch r_ptr.

Reply:
I'll try next a less invasive test with CI.